autocorpus/word_extractor.py

```diff
@@ -0,0 +1,165 @@
+    return tables
+
+
+def __convert_older_doc_file(file: Path, output_dir: Path) -> Path | bool:
+    """Converts an older .doc file to .docx format using platform-specific methods."""
+    operating_system = platform.system()
+    docx_path = Path(str(file).replace(".doc", ".docx"))
+    if operating_system == "Windows":
+        import win32com.client
+
+        word = None
+        try:
+            word = win32com.client.DispatchEx("Word.Application")
+            doc = word.Documents.Open(file)
+            doc.SaveAs(docx_path, 16)
+            doc.Close()
+            word.Quit()
+            return docx_path
+        except Exception:
+            return False
+        finally:
+            if word:
+                word.Quit()
+    elif operating_system == "Linux":
+        # Convert .doc to .docx using LibreOffice
+        subprocess.run(
+            [
+                "soffice",
+                "--headless",
+                "--convert-to",
+                "docx",
+                "--outdir",
+                output_dir,
+                file,
+            ],
+            check=True,
+            capture_output=True,
+        )
+        return docx_path
+    elif operating_system == "Darwin":  # macOS
+        try:
+            # AppleScript to open the file in Word and save as .docx
+            applescript = f'''
+            tell application "Microsoft Word"
+                open "{file}"
+                save as active document file name "{docx_path}" file format format document
+                close active document saving no
+            end tell
+            '''
+            subprocess.run(["osascript", "-e", applescript], check=True)
+            return docx_path
+        except Exception:
+            return False
+    else:
+        return False
+
+
+def extract_word_content(file_path: Path):
+    """Extracts text from a .doc file by converting it to .docx and processing with python-docx."""
+    if file_path.suffix.lower() not in [".doc", ".docx"]:
+        raise ValueError("Input file must be a .doc file.")
+    try:
+        output_dir = Path(file_path).parent.absolute()
+        # Check if the file is a .doc file
+        if file_path.suffix.lower() == ".doc":
+            docx_path = __convert_older_doc_file(file_path, output_dir)
+
+        # Extract text from the resulting .docx file
+        doc = Document(str(docx_path))
+        tables = __extract_tables(doc)
+        text_sizes = set(
+            [
+                int(x.style.font.size)
+                for x in doc.paragraphs
+                if x.style and x.style.font.size
+            ]
+        )
+        paragraphs = [
+            (
+                x.text,
+                True
+                if text_sizes
+                and x.style
+                and x.style.font.size
+                and int(x.style.font.size) > min(text_sizes)
+                else False,
+            )
+            for x in doc.paragraphs
+        ]
```

Comment:
It isn't obvious what a `list[tuple[str, bool]]` represents at a glance and I don't think it's documented where you use it either (e.g. in docstrings).

How about making a dataclass for these values? Then you can give the `str` and `bool` fields names, which should make it clearer.